graphql_compiler/interpreter/typedefs.py

```diff
@@ -172,8 +172,73 @@ def project_property(
         neighbor_hints: Optional[Collection[Tuple[EdgeInfo, NeighborHint]]] = None,
         **hints: Any,
     ) -> Iterable[Tuple[DataContext[DataToken], Any]]:
-        """Produce the values for a given property for each of an iterable of input DataTokens."""
-        # TODO(predrag): Add more docs in an upcoming PR.
+        """Produce the values for a given property for each of an iterable of input DataTokens.
+
+        In situations such as outputting property values or applying filters to properties,
+        the interpreter needs to get the value of some property field for a series of DataTokens.
+        For example, consider the following query:
+            {
+                Foo {
+                    bar @output(out_name: "bar_value")
+                }
+            }
+        Once the interpreter has used the get_tokens_of_type() function to obtain
+        an iterable of DataTokens for the Foo type, it will need to get the value of
+        the "bar" property on those tokens. This is where project_property() comes in.
+        The interpreter would call project_property() with arguments current_type_name = "Foo"
+        and field_name = "bar" along with an iterable of DataContext values, each of which
+        contains an optional DataToken as its current_token attribute. These DataContext
+        objects simply allow the interpreter to keep track of "which data came from where";
+        only the DataToken value within the current_token attribute is relevant to this function.
```

Comment:
```suggestion
        only the DataToken value bound to the current_token attribute is relevant to this function.
```
"Within" seems to imply an interstitial container--"bound to" or "referred to by" is probably clearer.

Reply:
Changed to match in several places.